### PR TITLE
Feat/trustedboot gcp new: Add build targets for gcp_trustedboot and gcp_trustedboot_gcp, corresponding platform tests + documentation

### PR DESCRIPTION
**Pull request by mguertler**
**What this PR does / why we need it**:

Implementing _trustedboot and _trustedboot_tpm2 builds and platform tests for the Google Cloud Platform

**Which issue(s) this PR fixes**:
Fixes internal #10 "Enable Secureboot images on GCP"

**Special notes for your reviewer**:

* Created new branch and redone changes from trustedboot_gcp due to complex git rebase issues, to keep commit history clean
* Content of this branch and trustedboot_gcp are equal
git diff remotes/origin/feat/trustedboot_gcp_new remotes/origin/feat/trustedboot_gcp
* branch feat/trustedboot_gcp successfully tested with nightly workflow
* Couldn't test this branch (feat/trustedboot_gcp_new) with nightly workflow during permission issues for this branch. However, this shouldn't be an issue due to the content equality of both branches.

**Release note**:

```feature developer
* Added gcp_trustedboot and gcp_trustedboot_tpm2 targets for automatic Github builds (nightly)
* Fixed & enhanced GCP platform tests, to properly test new build targets
* Fixed & enhanced documentation to describe GCP trustedboot features, custom builds & platform tests
```

### CODE REVIEW

**Comment by pnpavlov** [comment]
@nkraetzschmar, @yeoldegrove , could you please review the feature and tests?

File: .github/workflows/gcp_trustedboot_tests.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+#set -Eeuo pipefail
+
+# Name of Image to test
+cname="${@: -1}"
+
+configFile="gcp_test_config.yaml"
+containerName="ghcr.io/gardenlinux/gardenlinux/integration-test:today"
```
Comment (yeoldegrove) [inline .github/workflows/gcp_trustedboot_tests.sh:8]: Please update according to latest changes in `gcp_tets.sh`.
`containerName` changed.

File: .github/workflows/gcp_trustedboot_tests.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+artifact_dir="/tmp/gardenlinux-build-artifacts"
+platform_test_log_dir="/tmp/gardenlinux-platform-test-logs"
+
+mkdir -p "$platform_test_log_dir"
```
Comment (yeoldegrove) [inline .github/workflows/gcp_trustedboot_tests.sh:12]: Please update according to latest changes in `gcp_tets.sh`.
Can be removed.

File: .github/workflows/gcp_trustedboot_tests.sh [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      - cloud
+      - server
+      - base
+      - _slim
+      - _trustedboot
+EOF
+
+
+echo "### Start Integration Tests for gcp"
+podman run -it --rm -v "$(pwd):/gardenlinux" -v "$(dirname "$image_file"):/artifacts" -v "$platform_test_log_dir:/platform-test-logs" $containerName /bin/bash -s << EOF
```
Comment (yeoldegrove) [inline .github/workflows/gcp_trustedboot_tests.sh:84]: Please update according to latest changes in `gcp_tets.sh`.
`/platform-test-logs` is not needed anymore.

File: .github/workflows/gcp_trustedboot_tests.sh [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+mkdir /gardenlinux/tmp
+TMPDIR=/gardenlinux/tmp/
+cd /gardenlinux/tests
+export GOOGLE_APPLICATION_CREDENTIALS="/gardenlinux/$credentials_file_name"
+export CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE="/gardenlinux/$credentials_file_name"
+export GOOGLE_GHA_CREDS_PATH="/gardenlinux/$credentials_file_name"
+pytest --iaas=gcp --configfile=/gardenlinux/$configFile --junit-xml=/platform-test-logs/test-$cname-gcp_junit.xml || exit 1
```
Comment (yeoldegrove) [inline .github/workflows/gcp_trustedboot_tests.sh:91]: Please update according to latest changes in `gcp_tets.sh`.
The log file is now called `/artifacts/$cname.platform.test.xml`

File: .github/workflows/gcp_trustedboot_tpm2_tests.sh [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+# Name of Image to test
+cname="${@: -1}"
+
+configFile="gcp_test_config.yaml"
+containerName="ghcr.io/gardenlinux/gardenlinux/integration-test:today"
+artifact_dir="/tmp/gardenlinux-build-artifacts"
+platform_test_log_dir="/tmp/gardenlinux-platform-test-logs"
+
+mkdir -p "$platform_test_log_dir"
```
Comment (yeoldegrove) [inline .github/workflows/gcp_trustedboot_tpm2_tests.sh:12]: Please update according to latest changes in `gcp_tets.sh`.
See also `gcp_trustedboot_tests.sh`.

File: .github/workflows/gcp_trustedboot_tpm2_tests.sh [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+      - server
+      - base
+      - _slim
+      - _trustedboot
+      - _tpm2
+EOF
+
+
+echo "### Start Integration Tests for gcp"
+podman run -it --rm -v "$(pwd):/gardenlinux" -v "$(dirname "$image_file"):/artifacts" -v "$platform_test_log_dir:/platform-test-logs" $containerName /bin/bash -s << EOF
```
Comment (yeoldegrove) [inline .github/workflows/gcp_trustedboot_tpm2_tests.sh:85]: Please update according to latest changes in `gcp_tets.sh`.
See also `gcp_trustedboot_tests.sh`.

File: .github/workflows/gcp_trustedboot_tpm2_tests.sh [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+mkdir /gardenlinux/tmp
+TMPDIR=/gardenlinux/tmp/
+cd /gardenlinux/tests
+export GOOGLE_APPLICATION_CREDENTIALS="/gardenlinux/$credentials_file_name"
+export CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE="/gardenlinux/$credentials_file_name"
+export GOOGLE_GHA_CREDS_PATH="/gardenlinux/$credentials_file_name"
+pytest --iaas=gcp --configfile=/gardenlinux/$configFile --junit-xml=/platform-test-logs/test-$cname-gcp_junit.xml || exit 1
```
Comment (yeoldegrove) [inline .github/workflows/gcp_trustedboot_tpm2_tests.sh:92]: Please update according to latest changes in `gcp_tets.sh`.
See also `gcp_trustedboot_tests.sh`.

**Review by yeoldegrove — CHANGES_REQUESTED** [review]
I looked at the CI part of the PR. A few minor things need to be updated.

**Comment by mguertler** [comment]
Fixed gcp_*_tests.sh as requested

**Comment by mguertler** [comment]
_tpm2 ARM platform tests disabled, as they are disabled for AWS as well. It might be enabled, but has to be tested first.

> @mguertler Could you explain why you disable `gcp_trustedboot_tpm2` platform tests on `arm64`, please?
> 
> https://github.com/gardenlinux/gardenlinux/pull/2387/files#diff-1db27d93186e46d3b441ece35801b244db8ee144ff1405ca27a163bfe878957fR69-R70

**Review by yeoldegrove — COMMENTED** [review]
Looks OK from my side (platform test CI) now.
Final go should come from @nkraetzschmar

**Review by nkraetzschmar — APPROVED** [review]

**Review by yeoldegrove — APPROVED** [review]

**Pull request merged**